proxy.ts

```diff
@@ -0,0 +1,67 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+export async function proxy(request: NextRequest) {
+  // Skip API routes and static assets
+  if (request.nextUrl.pathname.startsWith('/_next') ||
+      request.nextUrl.pathname.startsWith('/api') ||
+      request.nextUrl.pathname.match(/\.(ico|png|jpg|jpeg|svg|css|js)$/)) {
+    return NextResponse.next();
+  }
+
+  const response = NextResponse.next();
+
+  try {
+    const drupalBaseUrl = process.env.NEXT_PUBLIC_DRUPAL_BASE_URL;
+    if (!drupalBaseUrl) {
+      return response;
+    }
+
+    const ceApiEndpoint = '/ce-api';
+    const path = request.nextUrl.pathname === '/' ? '/' : request.nextUrl.pathname.slice(1);
+
+    // Make a HEAD request to Drupal to get cache headers
+    // HEAD requests are lightweight as they don't return response body
+    const drupalResponse = await fetch(`${drupalBaseUrl}${ceApiEndpoint}/${path}`, {
```

Comment:
uhm, no we cannot do that. let's rather leave this feature out, but a 2nd request is a nogo